src/APL/vercel-apl.ts

```diff
@@ -65,23 +65,31 @@ export class VercelAPL implements APL {
   }
 
   private async saveDataToVercel(authData?: AuthData) {
-    debug(`saveDataToVercel with: ${authData}`);
+    debug("saveDataToVercel with: %j", authData);
+    let response: Response;
     try {
-      await fetch(this.registerAppURL, {
+      response = await fetch(this.registerAppURL, {
         method: "POST",
         headers: { "Content-Type": "application/json" },
         body: JSON.stringify({
           token: this.deploymentToken,
-          envs: {
-            [VercelAPLVariables.TOKEN_VARIABLE_NAME]: authData?.token || "",
-            [VercelAPLVariables.DOMAIN_VARIABLE_NAME]: authData?.domain || "",
-          },
+          envs: [
+            { key: VercelAPLVariables.TOKEN_VARIABLE_NAME, value: authData?.token || "" },
+            { key: VercelAPLVariables.DOMAIN_VARIABLE_NAME, value: authData?.domain || "" },
+          ],
         }),
       });
     } catch (error) {
       debug("Error during saving the data:", error);
-      throw new Error(`VercelAPL was not able to save auth data${error}`);
+      throw new Error(`VercelAPL was not able to save auth data ${error}`);
     }
+    if (response.status !== 200) {
```

Comment:
Its risky, technically Appraptor can return 202 and 204 (which will be valid here).

Safely to check if >=400 || <200